### ISSUE TITLE
expect("エ").toEqual("ア") does not throw Error

### ISSUE THREAD

**Issue by shigurenimo** [oven-sh/bun#14160]
### What version of Bun is running?

1.1.30

### What platform is your computer?

Darwin 24.0.0 arm64 arm

### What steps can reproduce the bug?

1. make empty directory
2. run `bun init`
3. make a test file `index.test.ts`
4. run `bun test`

![スクリーンショット 2024-09-26 1 11 42](https://github.com/user-attachments/assets/6029d0f4-0f1b-4e15-be93-703f455d61b8)

### What is the expected behavior?

should be

```bash
✗ toEqual [3.75ms]
3 | test("toEqual", () => {
4 |   expect("エ").toEqual("ア")
5 | })
6 | 
7 | test("toBe", () => {
8 |   expect("エ").toBe("ア")
                  ^
error: expect(received).toBe(expected)

Expected: "ア"
Received: "エ"
```

### What do you see instead?

```ts
import { test, expect } from "bun:test"

test("toEqual", () => {
  expect("ア").toEqual("ア")
})

test("toBe", () => {
  expect("a").toBe("ア")
})
```

```bash
$ bun test
bun test v1.1.30-canary.37 (117e1b38)

index.test.ts:
✓ toEqual
✓ toBe [1.04ms]

 2 pass
 0 fail
 2 expect() calls
Ran 2 tests across 1 files. [9.00ms]
```

### Additional information

I had to use Katakana in my project. the cause is Katakana?

```ts
import { test, expect } from "bun:test"

test("Katakana and Alphabet", () => {
  expect("ア").toEqual("a") // fail
})

test("Alphabet and Katakana", () => {
  expect("a").toEqual("ア") // fail
})

test("Katakana and Katakana", () => {
  expect("フ").toEqual("ア") // pass
})

test("Kanji and Katakana", () => {
  expect("了").toEqual("ア") // fail
})
```

**Comment by dylan-conway** [comment]
This is most likely bug in our diff formatter. It is appending invalid UTF-8 sequences, which will error when we try to create the error message with `WTF::String::fromUTF8`, which will then fail to create the error for `toEqual`

**Comment by reiwa** [comment]
to avoid this problem, it's possible to use an array for now.

```ts
expect(["エ"]).toEqual(["ア"])
```

**Comment by pfgithub** [comment]
This was fixed in 1.1.31
